### PR TITLE
Fixed issue #16042  generated token length is too short with add participants()

### PR DESCRIPTION
**Pull request by gabrieljenik**
Updated code from generateTokens()
Code reorg

### CODE REVIEW

File: application/models/Token.php [hunk 1]
```diff
@@ -61,6 +61,9 @@ abstract class Token extends Dynamic
     /** @var int Maximum token length */
     const MAX_LENGTH = 36;
 
+    /** @var int Default token length */
+    const DEFAULT_LENGTH = 15;
```
Comment (Shnoulle) [inline application/models/Token.php:65]: We don't have a config for this ? I think yes, but no ....

File: application/models/Token.php [hunk 2]
```diff
@@ -231,13 +234,34 @@ public function findByToken($token)
...
+    }
+
     /**
      * Generates a token for this object.
      * @throws CHttpException
      */
     public function generateToken($tokenlength = NULL)
     {
-        $iTokenLength = $tokenlength ? $tokenlength : $this->survey->tokenlength;
+        if ($tokenlength) {
```
Comment (Shnoulle) [inline application/models/Token.php:260]: Why not `generateToken($iTokenLength = NULL)`

And 
```
if(empty($iTokenLength)) {
    $iTokenLength = $this->getSurveyTokenLength();
}
```

I don't understand why use 2 different parameter name here ....

Comment (gabrieljenik) [reply application/models/Token.php]: You are right.

File: application/models/Token.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/Token.php:65]: Would be the best, but we don't have it and I just wanted to wrap the ticket :)
The less changes, the less tests, the less destabilization

Comment (Shnoulle) [reply application/models/Token.php:65]: totally right :+1:

**Review by Shnoulle — APPROVED** [review]

**Review by eddylackmann — APPROVED** [review]

**Pull request merged**